pkg/portal/graphql/app_mutation.go

```diff
@@ -64,6 +64,12 @@ var _ = registerMutationField(
 			appID := resolvedNodeID.ID
 
 			gqlCtx := GQLContext(p.Context)
+
+			err := gqlCtx.AuthzService.CheckAccessOfViewer(appID)
```

Comment:
I think this check should be done in `loader`/`service` layer (i.e. within `gqlCtx.Apps.Get` call), otherwise it's easy to miss the checks when authoring new query/mutation.

Reply:
Service is at the building build level so it may not have enough context to determine access control. Let me review whether we can do it at loader layer.